### PR TITLE
Surge example

### PR DESCRIPTION
**Pull request by zzstoatzz**
adds example using https://surgemsg.com/ to text myself from claude

```console
fastmcp install examples/text_me.py -f .env && fastmcp run examples/text_me.py
```

also updates tests to only run on changes to src, tests, and dep-related files

### CODE REVIEW

File: examples/text_me.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# /// script
+# dependencies = ["fastmcp"]
+# ///
+
+"""
+FastMCP Echo Server
```
Comment (jlowin) [inline examples/text_me.py]:
```suggestion
FastMCP Surge Server
```

File: examples/text_me.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# /// script
+# dependencies = ["fastmcp"]
```
Comment (jlowin) [inline examples/text_me.py:2]: This *should* be unecessary as fastmcp always adds itself as a dependency but 👍 good practice. I did learn that if you have a script declaration like this it overrules all other dependencies though

**Review by jlowin — APPROVED** [review]
Thanks! Do you mind adding some brief documentation to the top of the example file - maybe what env vars to set or how to deploy with env

**Comment by zzstoatzz** [comment]
good calls @jlowin - added in [cbb8dbe](https://github.com/jlowin/fastmcp/pull/29/commits/cbb8dbeccfa62479900d12f38cff6267cea61249)

**Pull request merged**